### ISSUE TITLE
FallbackPolicy should jump in when the top n intents have similar confidence

### ISSUE THREAD

**Issue by akelad** [RasaHQ/rasa_core#1574]
Right now it only jumps in if the NLU confidence is below a certain threshold. It should however also jump in if the top 2 intents have a confidence of 0.98 and 0.80 for example.
I've already started looking into this for the rasa-demo

**Comment by amn41** [comment]
For another project we also saw `highest confidence` - `second highest confidence` was a better metric than absolute value, so yes this should be configurable in the fallbackpolicy.

**Comment by amn41** [comment]
thinking about it now @akelad I think it makes more sense if NLU lets you configure how the  confidence is calculated. wdyt?

**Comment by akelad** [comment]
hmm, does it? surely NLU just returns the intent ranking and then you deal with that on the core side?

**Comment by amn41** [comment]
I think a lot of the time `highest confidence` - `second highest confidence` is a better metric to use for "confidence" than the rank of the highest scoring intent. So even if not using Core, still makes sense to configure that. Wdyt ?

**Comment by akelad** [comment]
Yeah I def agree that `highest confidence - second highest confidence` is a better metric. That should also be applied to third highest confidence, because sometimes the top x are very similar
